File: cli/src/config.test.ts
import { describe, expect, test } from 'bun:test';
import { splitTomlFrontmatter, sourcePath, validateContentEntry } from './config.js';

describe('config parsing helpers', () => {
  test('splitTomlFrontmatter without frontmatter', () => {
    const result = splitTomlFrontmatter('hello world');
    expect(result.hasFrontmatter).toBeFalse();
    expect(result.body).toBe('hello world');
  });

  test('splitTomlFrontmatter with frontmatter', () => {
    const result = splitTomlFrontmatter('+++\nkey = "value"\n+++\nbody');
    expect(result.hasFrontmatter).toBeTrue();
    expect(result.frontmatter).toBe('key = "value"');
    expect(result.body).toBe('body');
  });

  test('sourcePath prefers source', () => {
    expect(
      sourcePath({
        source: 'docs/*.md',
        path: '',
        targets: [],
        output: '',
        exclude: [],
        preserve: [],
        frontmatter: '',
        prompt: '',
        check_cmd: '',
        check_cmds: {},
      }),
    ).toBe('docs/*.md');
  });

  test('validateContentEntry requires output when targets exist', () => {
    expect(() =>
      validateContentEntry({
        source: 'docs/*.md',
        path: '',
        targets: ['es'],
        output: '',
        exclude: [],
        preserve: [],
        frontmatter: '',
        prompt: '',
        check_cmd: '',
        check_cmds: {},
      }),
    ).toThrow();
  });
});

File: cli/internal/glossia/config_test.go
package glossia

import "testing"

func TestSplitTomlFrontmatter_NoFrontmatter(t *testing.T) {
	res, err := splitTomlFrontmatter("hello world")
	if err != nil {
		t.Fatalf("unexpected error: %v", err)
	}
	if res.HasFrontmatter {
		t.Fatalf("expected HasFrontmatter=false")
	}
	if res.Body != "hello world" {
		t.Fatalf("unexpected body: %q", res.Body)
	}
}

func TestSplitTomlFrontmatter_WithFrontmatter(t *testing.T) {
	res, err := splitTomlFrontmatter("+++\nkey = \"value\"\n+++\nbody")
	if err != nil {
		t.Fatalf("unexpected error: %v", err)
	}
	if !res.HasFrontmatter {
		t.Fatalf("expected HasFrontmatter=true")
	}
	if res.Frontmatter != "key = \"value\"" {
		t.Fatalf("unexpected frontmatter: %q", res.Frontmatter)
	}
	if res.Body != "body" {
		t.Fatalf("unexpected body: %q", res.Body)
	}
}

func TestSourcePath_PrefersSource(t *testing.T) {
	got := sourcePath(ContentEntry{
		Source:  "docs/*.md",
		Path:    "",
		Targets: []string{},
	})
	if got != "docs/*.md" {
		t.Fatalf("unexpected sourcePath: %q", got)
	}
}

func TestValidateContentEntry_RequiresOutputWhenTargetsExist(t *testing.T) {
	err := validateContentEntry(ContentEntry{
		Source:  "docs/*.md",
		Path:    "",
		Targets: []string{"es"},
		Output:  "",
	})
	if err == nil {
		t.Fatalf("expected error")
	}
}
